### PR TITLE
feat: change say() method response type from void to Message

### PR DESCRIPTION
**Pull request by su-chang**
## I'm submitting a...

```
[ ] Bug Fix
[ ] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

_please describe the changes that you are making_

_for features, please describe how to use the new feature_

_please include a reference to an existing issue, if applicable_

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by su-chang** [comment]
related issue: #1857

**Comment by huan** [comment]
@windmemory Please only approve the PR that can be passed the CI tests.

It will not be good progress that we approve of the future work before we can make sure the current work can be done without any problem.

This PR can not pass the CI before the new version of wechaty-puppet had published the related new features, and the wechaty-puppet is not ready yet. So please only approve this PR after we had everything ready for it.

Thank you very much!

**Comment by windmemory** [comment]
@huan got it
:P

**Comment by windmemory** [comment]
Let's make the `codeclimate` happy first.

File: src/user/contact.ts [hunk 1]
```diff
@@ -404,7 +404,7 @@ export class Contact extends Accessory implements Sayable {
     } else {
       throw new Error('unsupported arg: ' + something)
     }
-    if (msgId) {
+    if (typeof msgId === 'string') {
```
Comment (huan) [inline src/user/contact.ts]:
```ts
    if (typeof msgId === 'string' && msgId) {
```

**Comment by huan** [comment]
BTW: It would be better if you can remove the code of `typeof msgId === 'string'` because it's useless: the Typing system will be able to take care of that case.

**Comment by su-chang** [comment]
```js
@param {(string | Contact | FileBox | UrlLink | MiniProgram)} something
send text, Contact, or file to contact. </br>
You can use {@link https://www.npmjs.com/package/file-box|FileBox} to send file
@returns {Promise<void | Message>}

import { FileBox }  from 'file-box'
const fileBox1 = FileBox.fromUrl('https://chatie.io/wechaty/images/bot-qr-code.png')
const fileBox2 = FileBox.fromFile('/tmp/text.txt')
await contact.say(fileBox1)
const msg1 = await contact.say(fileBox1) // only supported by puppet-padplus
await contact.say(fileBox2)
const msg2 = await contact.say(fileBox2) // only supported by puppet-padplus
```

Should I add JS doc like this way?

Or declare `only supported by puppet-padplus` only once in head line?

```js
@param {(string | Contact | FileBox | UrlLink | MiniProgram)} something
send text, Contact, or file to contact. </br>
You can use {@link https://www.npmjs.com/package/file-box|FileBox} to send file
@returns {Promise<void | Message>} Message type only supported by puppet-padplus.

import { FileBox }  from 'file-box'
const fileBox1 = FileBox.fromUrl('https://chatie.io/wechaty/images/bot-qr-code.png')
const fileBox2 = FileBox.fromFile('/tmp/text.txt')
await contact.say(fileBox1)
const msg1 = await contact.say(fileBox1)
await contact.say(fileBox2)
const msg2 = await contact.say(fileBox2)
```

Which way is better?  Thanks.
@huan

**Comment by huan** [comment]
Please just pick the one which you think is the best. Thanks.

**Comment by windmemory** [comment]
Looks good to me.

**Comment by huan** [comment]
Thank you very much for the great improvement, Cheers!

**Pull request merged**